pkg/nodeconfig/payload/payload.go

```diff
@@ -82,3 +84,39 @@ func NewFileInfo(path string) (*FileInfo, error) {
 		SHA256: fmt.Sprintf("%x", sha256.Sum256(contents)),
 	}, nil
 }
+
+// Validate ensures all required payload files exist
+func Validate() error {
```

Comment:
Do you anticipate more validations to be included here? Otherwise, I don't feel the `Validate()` function is needed just to hold the collection of files. You can call directly `checkIfFilesExist`.